StaticAnalyzer/views/android/manifest_analysis.py

```diff
@@ -1260,8 +1260,21 @@ def read_manifest(app_dir, tools_dir, typ, binary):
         manifest = ''
         if binary is True:
             print("[INFO] Getting Manifest from Binary")
-            print("[INFO] AXML -> XML")
-            manifest = os.path.join(app_dir, "AndroidManifest.xml")
+            manifest = os.path.join(app_dir, "apktool_out", "AndroidManifest.xml")
+            if isFileExists(manifest):
+                print("[INFO] using Manifest from apktool")
+                with io.open(
+                    manifest,
+                    mode='r',
+                    encoding="utf8",
+                    errors="ignore"
+                ) as file_pointer:
+                    dat = file_pointer.read()
+                return dat
+            else:
+                print("[INFO] Using Manifest from Binary")
+                print("[INFO] AXML -> XML")
+                manifest = os.path.join(app_dir, "AndroidManifest.xml")
             if len(settings.AXMLPRINTER_BINARY) > 0 and isFileExists(settings.AXMLPRINTER_BINARY):

```

Comment:
What is the scope of this else ? In this case both code using apktool and axmlprinter executes as if at line 1278 is in outer scope. We don't need both to execute, right?

Reply:
you have a return on line 1273 if apktool_out/ contain a Manifest 

will review the code, Cause I thins the if in 1278 should be "idented" better  part of the else on 1274 

good catch @ajinabraham